ts/Core/Axis/Axis.ts

```diff
@@ -2168,26 +2182,84 @@ class Axis {
                         otherOptions.height,
                         otherOptions.pane
                     ].join(',');
 
-
-                if (axis.series.length) { // #4442
-                    if ((others as any)[key]) {
+                if (
+                    // #4442
+                    series.length &&
+                    series.some((s): boolean => s.visible)
+                ) {
+                    if (others[key]) {
                         hasOther = true; // #4201
+                        alignedAxes.push(otherAxis);
                     } else {
-                        (others as any)[key] = 1;
+                        others[key] = 1;
+                    }
+
+                    // Handle alignThresholds. The `thresholdAlignments` array
+                    // keeps records of where each axis in the group wants its
+                    // threshold, from 0 which is on `axis.min`, to 1 which is
+                    // on `axis.max`.
+                    if (alignThresholds) {
+                        if (
+                            !isNumber(otherAxis.dataMin) ||
+                            (
+                                otherAxis !== axis &&
+                                series.some((s): boolean|undefined => (
+                                    s.isDirty || s.isDirtyData
+                                ))
+                            )
+                        ) {
+                            otherAxis.getSeriesExtremes();
+                        }
+                        let thresholdAlignment = clamp(
+                            (
+                                (
+                                    (otherAxis.threshold || 0) -
+                                    (otherAxis.dataMin || 0)
+                                ) /
+                                (
+                                    (otherAxis.dataMax || 0) -
+                                    (otherAxis.dataMin || 0)
+                                )
+                            ),
+                            0,
+                            1
+                        );
+                        if (otherOptions.reversed) {
+                            thresholdAlignment = 1 - thresholdAlignment;
+                        }
+                        thresholdAlignments.push(thresholdAlignment);
                     }
                 }
             });
         }
+
+        if (hasOther) {
+            // For each of the axes in the group, record the average
+            // `thresholdAlignment`.
+            const thresholdAlignment = thresholdAlignments.length > 1 ?
+                thresholdAlignments.reduce(
+                    (sum: number, n): number => (sum += n),
+                    0
+                ) / thresholdAlignments.length :
+                void 0;
+
+            alignedAxes.forEach((axis): number|undefined => (
+                axis.thresholdAlignment = thresholdAlignment
+            ));
```

Comment:
I was a little bit confused about this arrow function's return. It's working as expected, but semantically forEach shouldn't return a number type.

What about a standard (less confusing) loop?

alignedAxes.forEach((axis): void => {
    axis.thresholdAlignment = thresholdAlignment;
});

The same with `append` and `prepend` functions below.

Reply:
Hah, just yesterday had the same problem. Arrow functions by definition return something, so TS requires us to define what should be returned. I have decided to use `function () {}` instead 🤷

Reply:
It could still be an arrow function, but with curly brackets (without an immediate return in the first line). Then it just returns `undefined` (like in my proposal)

Reply:
Oh, right! Missed that part, sorry

Reply:
Yep, code like this might be confusing :D